src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
 
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K, V> node = table[hash];
+
+        while (node != null) {
+            if (node.hash == hash
+                    && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % DEFAULT_CAPACITY);
```

Comment:
Bug: The way you calculate the hash is incorrect. You are using the `DEFAULT_CAPACITY` as part of the hash calculation, which is not correct. The hash should only depend on the key, not on the capacity of the table. Moreover, when you resize the table, this hash calculation will lead to incorrect results because it still uses the initial capacity. You should calculate the hash based on the key only, and then use this hash to calculate the index in the table array by doing `hash % table.length`.